src/Sushi.php

```diff
@@ -2,90 +2,91 @@
 
 namespace Sushi;
 
+use DateTime;
 use Illuminate\Database\Connectors\ConnectionFactory;
-use Illuminate\Support\Str;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\SQLiteConnection;
+use Illuminate\Support\Arr;
+use Illuminate\Support\LazyCollection;
 
+/** @mixin Model */
 trait Sushi
 {
-    protected static $sushiConnection;
+    protected static SQLiteConnection $sushiConnection;
 
-    public static function resolveConnection($connection = null)
+    protected static function bootSushi(): void
     {
+        static::migrate();
+    }
+
+    public function getConnection(): SQLiteConnection
+    {
+        static::$sushiConnection ??= $this->getConnectionFactory()->make([
```

Comment:
`??=` is PHP 7.4 only :-(

Reply:
Right, that's why I've added it to the `composer.json` - for me it's never a problem to require the latest PHP version a month or two after it's released. But there's a longer syntax, so if PHP7.3 or below should be supported it's possible to rewrite.

Reply:
And both are valid points/opinions. Because I run PHP7.4 and this PR is only a draft - based on my customizations - I've chosen to use the new 7.4 features and restrict the minimum version.
If anything of this PR should make it in this repo and Caleb doesn't want to restrict I'm fine with removing this sugar and do it in PHP7.1/7.2 style.